### PR TITLE
Add audit check for URL schema

### PR DESCRIPTION
**Pull request by Git-Jiro**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [X] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [ ] Have you successfully run `brew tests` with your changes locally?

-----
This implements: https://github.com/caskroom/homebrew-cask/issues/25403

### CODE REVIEW

File: Library/Homebrew/cask/lib/hbc/audit.rb [hunk 1]
```diff
@@ -243,6 +244,78 @@ def core_formula_url
...
+      #   0, nil           : URL unreachable
+      #  -1, effective URL : URL reachable, but schema changed
+
+      curl_executable, *args = curl_args(
+        "--compressed", "--location", "--fail",
+        "--write-out", "%{http_code} %{url_effective}",
+        "--output", "/dev/null",
+        url_to_access,
+        user_agent: :fake
+      )
```
Comment (reitermarkus) [inline Library/Homebrew/cask/lib/hbc/audit.rb]: This will download the whole file to `/dev/null`, won't it? It should use `--head`.

**Comment by MikeMcQuaid** [comment]
It would be good to share more code here with `audit`'s equivalent.

**Comment by Git-Jiro** [comment]
@MikeMcQuaid Ok, I will have a look.

**Comment by Git-Jiro** [comment]
@MikeMcQuaid is this the kind of code reuse you envisioned?

**Comment by MikeMcQuaid** [comment]
@Git-Jiro Yes, something like that. You may want to extract the `FormulaAuditor.check_http_content` into `utils/curl` instead as importing a Homebrew `cmd/audit` inside Cask seems a little weird. Wait for @reitermarkus's thoughts first, though.

**Comment by Git-Jiro** [comment]
@MikeMcQuaid Do you have an example for me where I can look at a similar refactoring? I would like to see how other people handled similar situations in the code base.

**Comment by MikeMcQuaid** [comment]
@Git-Jiro I don't have a good example I can think of but I can answer more specific questions and provide review feedback.

**Comment by Git-Jiro** [comment]
Ok

**Comment by Git-Jiro** [comment]
@MikeMcQuaid What do you think about my first version of this refactoring?

**Comment by MikeMcQuaid** [comment]
@Git-Jiro Looks great, thanks for that! I'm happy to 🚢 this when @reitermarkus is.

File: Library/Homebrew/cask/lib/hbc/audit.rb [hunk 2]
```diff
@@ -275,6 +277,17 @@ def core_formula_url
       "#{core_tap.default_remote}/blob/master/Formula/#{cask.token}.rb"
     end
 
+    def check_https_availability
+      check_url_for_https_availability(cask.url) unless cask.url.to_s.empty?
```
Comment (reitermarkus) [inline Library/Homebrew/cask/lib/hbc/audit.rb]: This should probably also use `cask.url.user_agent`.

File: Library/Homebrew/cask/lib/hbc/audit.rb [hunk 3]
```diff
@@ -275,6 +277,18 @@ def core_formula_url
       "#{core_tap.default_remote}/blob/master/Formula/#{cask.token}.rb"
     end
 
+    def check_https_availability
+      user_agents = cask.url.user_agent.to_s.empty? ? [:default] : [cask.url.user_agent]
```
Comment (reitermarkus) [inline Library/Homebrew/cask/lib/hbc/audit.rb]: `#user_agent` already defaults to `:default` by itself.

**Comment by reitermarkus** [comment]
Thanks, @Git-Jiro, nice work here!

**Pull request merged**